src/clr-angular/forms/combobox/options.ts

```diff
@@ -1,59 +1,140 @@
 /**
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component, ElementRef, Inject, Injector, OnDestroy, Optional } from '@angular/core';
-import { Subscription } from 'rxjs';
-
-import { AbstractPopover } from '../../popover/common/abstract-popover';
-import { Point } from '../../popover/common/popover';
+import {
+  Component,
+  ContentChildren,
+  ElementRef,
+  Inject,
+  Input,
+  Optional,
+  QueryList,
+  AfterViewInit,
+  OnDestroy,
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
 import { POPOVER_HOST_ANCHOR } from '../../popover/common/popover-host-anchor.token';
+import { OptionSelectionService } from './providers/option-selection.service';
+import { IF_ACTIVE_ID } from '../../utils/conditional/if-active.service';
+import { AriaService } from '../../utils/aria/aria.service';
+import { ClrCommonStringsService } from '../../utils/i18n/common-strings.service';
+import { LoadingListener } from '../../utils/loading/loading-listener';
+import { ClrLoadingState } from '../../utils/loading/loading';
+import { ClrOption } from './option';
+import { ComboboxFocusHandler } from './providers/combobox-focus-handler.service';
+import { fromEvent, Subscription } from 'rxjs';
+import { ClrPopoverToggleService } from '../../utils/popover/providers/popover-toggle.service';
+
+let nbOptionsComponents = 0;
+
+@Component({
+  selector: 'clr-options',
+  template: `
+    <div *ngIf="optionSelectionService.loading" class="clr-options-loading">
+      <clr-spinner clrInline>
+        Loading...
+      </clr-spinner>
+      <span class="clr-options-text">
+        {{ searchText(optionSelectionService.currentInput) }}
+      </span>
+    </div>
 
-@Component({ selector: 'clr-options', templateUrl: './options.html', host: { '[class.clr-options]': 'true' } })
-export class ClrOptions extends AbstractPopover implements OnDestroy {
-  private sub: Subscription;
+    <!-- Rendered if data set is empty -->
+    <div *ngIf="emptyOptions">
```

Comment:
I hadn't ever thought of using a QueryList like this, neat!

I think it just needs a test that the correct things get set when a a new ClrOptions component is created (and the gettor is returning the correct thing). 

I would like to see that if there is an *ngIf on one of the options the QueryList rflects the correct ClrOptions. I'm not 100% sure about this as it might just be retesting the QueryList behavior so you your discresion here 